### DISCUSSION TITLE
doubts

### DISCUSSION THREAD

**Discussion by harishgovindarajan** [karrioapi/karrio#509] [General]
Previously, with the test data mode enabled, I could check the current status of my package. However, now it's displaying 'authentication failed.' Has anyone else encountered this issue, and does anyone have insights on how to resolve it?

While in test data mode, I noticed that when I select a carrier, there is a suffix named '(sandbox)' that appears. Can anyone kindly explain what this suffix represents and why it is showing? Your insights would be much appreciated. Thank you!

I'm struggling to understand the documentation process for adding a custom carrier. Can anyone provide a step-by-step guide on how to add a custom carrier? Your assistance would be greatly appreciated. Thank you!

**Comment by danh91** [comment]
Hi @harishgovindarajan ,

- Can you tell me which carrier you had the tracking issue with? The ‘authentication failed’ error, does it only appear when you make a tracking request?

- Regarding the (Sandbox) suffix, the idea was to be an indicator of carriers available in test mode. I agree that it can be confusing. (Should be removed in upcoming release)


- As for the custom carrier integration, it is like writing a plugin in Python for the carrier you want to add. I agree the docs are limited. I am hoping to do a video walkthrough as well. What I recommend right now is to get started with the instructions here   
https://docs.karrio.io/carriers/sdk/extension and push a branch or a PR. 
There are many carrier plugins that you can take inspiration from if you glance at the code.
You can DM me on Discord or post on our main channel if anything is confusing or blocking and we will be able to help.
